core/store/src/trie/update.rs

```diff
@@ -126,7 +122,9 @@ impl TrieUpdate {
         self.prospective.clear();
     }
 
-    pub fn finalize(self) -> Result<(TrieChanges, Vec<RawStateChangesWithTrieKey>), StorageError> {
+    pub fn finalize(
```

Comment:
As we change it - can we add a comment saying that it prepares state changes to apply to storage in the future and returns underlying `Trie` back?
Maybe it is obvious for most people because it is called `finalize` and takes `self`, but it never was the case for me.